### PR TITLE
Build Ceres, SuiteSparse, Boost and Eigen outside of OpenMVG in Travis

### PR DESCRIPTION
**Pull request by caymard**
To avoid Travis build Ceres at each push, it is build before OpenMVG and put it in Travis cache system, like it's already done with OpenCV.

Same thing has been done for SuiteSparse, Boost and Eigen.

connected to poparteu/openMVG#51

### CODE REVIEW

**Comment by simogasp** [comment]
what's the status?

**Comment by caymard** [comment]
travis script is now configured to build OpenMVG with external ceres/eigen/boost. So we're good.

File: .travis.yml [hunk 1]
```diff
@@ -57,24 +70,106 @@ before_install:
  - mkdir --parent $OPENCV_CONTRIB
  - mkdir --parent $OPENCV_BUILD
  - mkdir --parent $OPENCV_INSTALL
- - if [ ! "$(ls -A $OPENCV_INSTALL)" ]; then git clone --recursive --branch 3.0.0 --depth 1 https://github.com/Itseez/opencv.git $OPENCV_SOURCE; fi
- - if [ ! "$(ls -A $OPENCV_CONTRIB)" ]; then git clone --branch 3.0.0 --depth 1  https://github.com/Itseez/opencv_contrib.git $OPENCV_CONTRIB; fi
+ - >
+    if [ ! "$(ls -A $OPENCV_INSTALL)" ]; then
```
Comment (fabiencastan) [inline .travis.yml]:
```
if [ ! -d "$DIRECTORY" ]; then
  # Control will enter here if $DIRECTORY doesn't exist.
fi
```
This avoid warnings.

Comment (fabiencastan) [inline .travis.yml]: It's better to put an explicit message:
```
if [ -d "$DIRECTORY" ]; then
  echo "Use OpenCV pre-builded cache."
else
  echo "Build OpenCV (not in cache):"
  # Build instructions...
fi
```

**Pull request merged**